numba/tests/test_dispatcher.py

```diff
@@ -1003,846 +949,6 @@ def pyfunc(A, i, j):
-
-    # The source file that will be copied
-    usecases_file = None
-    # Make sure this doesn't conflict with another module
-    modname = None
-
-    def setUp(self):
-        self.tempdir = temp_directory('test_cache')
-        sys.path.insert(0, self.tempdir)
-        self.modfile = os.path.join(self.tempdir, self.modname + ".py")
-        self.cache_dir = os.path.join(self.tempdir, "__pycache__")
-        shutil.copy(self.usecases_file, self.modfile)
-        self.maxDiff = None
-
-    def tearDown(self):
-        sys.modules.pop(self.modname, None)
-        sys.path.remove(self.tempdir)
-
-    def import_module(self):
-        # Import a fresh version of the test module.  All jitted functions
-        # in the test module will start anew and load overloads from
-        # the on-disk cache if possible.
-        old = sys.modules.pop(self.modname, None)
-        if old is not None:
-            # Make sure cached bytecode is removed
-            cached = [old.__cached__]
-            for fn in cached:
-                try:
-                    os.unlink(fn)
-                except FileNotFoundError:
-                    pass
-        mod = import_dynamic(self.modname)
-        self.assertEqual(mod.__file__.rstrip('co'), self.modfile)
-        return mod
-
-    def cache_contents(self):
-        try:
-            return [fn for fn in os.listdir(self.cache_dir)
-                    if not fn.endswith(('.pyc', ".pyo"))]
-        except FileNotFoundError:
-            return []
-
-    def get_cache_mtimes(self):
-        return dict((fn, os.path.getmtime(os.path.join(self.cache_dir, fn)))
-                    for fn in sorted(self.cache_contents()))
-
-    def check_pycache(self, n):
-        c = self.cache_contents()
-        self.assertEqual(len(c), n, c)
-
-    def dummy_test(self):
-        pass
-
-
-class BaseCacheUsecasesTest(BaseCacheTest):
-    here = os.path.dirname(__file__)
-    usecases_file = os.path.join(here, "cache_usecases.py")
-    modname = "dispatcher_caching_test_fodder"
-
-    def run_in_separate_process(self):
-        # Cached functions can be run from a distinct process.
-        # Also stresses issue #1603: uncached function calling cached function
-        # shouldn't fail compiling.
-        code = """if 1:
-            import sys
-
-            sys.path.insert(0, %(tempdir)r)
-            mod = __import__(%(modname)r)
-            mod.self_test()
-            """ % dict(tempdir=self.tempdir, modname=self.modname)
-
-        popen = subprocess.Popen([sys.executable, "-c", code],
-                                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        out, err = popen.communicate()
-        if popen.returncode != 0:
-            raise AssertionError(
-                "process failed with code %s: \n"
-                "stdout follows\n%s\n"
-                "stderr follows\n%s\n"
-                % (popen.returncode, out.decode(), err.decode()),
-            )
-
-    def check_module(self, mod):
```

Comment:
@gmarkall from OOB discussion, this was the function that we established did nothing/was never run?

Reply:
It is.

Reply:
Thanks, its removal makes sense!